### DISCUSSION TITLE
Automated documentation

### DISCUSSION THREAD

**Discussion by CM000n** [UI-Lovelace-Minimalist/UI#100] [Ideas]
Hi guys, because the repository is picking up speed again and more and more pull requests with new features and changes are coming in, it is getting more and more difficult to update the documentation / wiki page in a timely manner.
On top of that, due to @Paddy0174 outage, we currently don't have access permissions to the [wiki page repo](https://github.com/UI-Lovelace-Minimalist/page) and more and more necessary adjustments are piling up there.
Therefore I recently asked myself if it would be possible to automatically generate maps and documentation from the latest .yaml files. Also, @sildehoop last post about his [Figma](https://github.com/UI-Lovelace-Minimalist/UI/discussions/83) made me wonder if such a card display and associated documentation could be automated somehow.
This is certainly a bigger and more complex issue. Unfortunately I had no bright idea how to implement this. Therefore I simply put the topic here for discussion. Maybe someone here can and might support.

**Comment by KaherdinTristan** [comment]
Hi
I love the fact that this repo has "exploded", because the results are amazing and the community interaction is so cool 💙🤠

The new cards really deserve a place to be shown, one by one, so that new and existing users can see all options available.

Has GitHub heard back regarding the wiki?
🕊️

**Reply by CM000n** [reply]
Yes, I just received feedback from Github support this morning. I do not want to withhold this from you. Unfortunately, it looks like we can't get permissions to the wiki repo or organization. 😌 So the general question remains; How do we handle this organsisation created by @Paddy0174?



```
Notch (GitHub Support)
Dec 22, 2021, 3:02 AM UTC

Hello Simon,
Thanks for reaching out to GitHub Support.Only the owner of the organization can make such administrative changes as adding collaborators to repositories on the account. 
For security reasons, GitHub Support cannot grant you ownership to another individual's repositories, even if you are a collaborator on another repository of theirs.
If you're unable to contact the owner directly, we can relay a message on your behalf one single time. 
If that sounds good, please reply with the exact message you'd like us to pass along. 
Please be sure to identify yourself in a way they'll be able to recognise and to provide your preferred contact details for us to share.
I look forward to hearing back from you.

NotchGitHub Support
```

**Reply by CM000n** [reply]
FYI: @TBens @schumijo, @bavo

**Reply by KaherdinTristan** [reply]
Could you clone the repo, and start a new base of operations? 🥺

**Reply by stokkie90** [reply]
I agree with @KaherdinTristan, if we start with a fresh GitHub organization  we can make sure multiple people have owner permissions to prevent this in the future. And it could attract even more people as the Wiki/Web page will be up-to-date with new card additions and in line with the UI repo.

**Reply by sildehoop** [reply]
I think this is our only option if we want to update the wiki short term

**Reply by stokkie90** [reply]
Just an idea. Why not integrate the docs/wiki into this repo. All in one location.

This allows to add documentation and changes in one PR. just create a `docs` folder and choose a static framework to build on. 

Currently it's build on Jekyll but we could swap it for something else like [mkdocs](https://squidfunk.github.io/mkdocs-material)

**Reply by bavo** [reply]
I'm also in favor of pulling the documentation into this repository. I was really confused the first time i was looking for the documentation because there were some links pointing to the wrong location and i wanted to fix them.
Took me a long time to figure out that the documentation was somewhere separate.

**Reply by stokkie90** [reply]
Wasn't that hard 😅 

https://github.com/UI-Lovelace-Minimalist/UI/pull/137

https://gallant-einstein-431c17.netlify.app/

**Reply by stokkie90** [reply]
Maybe we should rethink the current repo layout if we want to support things like HACS. I was looking at dwains_dashboard and it seems it's advertising it as an [Integration](https://hacs.xyz/docs/publish/integration) That will involve some python coding. but looking at the code, it shouldn't be that hard to make it work. I got workout some repo layout that will support:

- HACS
- Documentation/wiki
  - Optionally: Option to share dashboards 
     - we could also use Discussions for this. 

Are there some more things we would like to see / improve. 

I think if we work together on this we can make something beautiful. I'm not that experienced in designing cards & themes. but i can help out in writing python code and some automations / ci testing.

**Reply by CM000n** [reply]
Wow, that was super fast. Thanks for the input @stokkie90. 
I have been thinking about reframing the repo myself. I think we will definitely need it in the future. I just don't know yet if it makes sense to keep running this repo if we don't have access to the organisation it hangs under.
However, I saw today that @Paddy0174 was online in the Home Assistant forum yesterday. I have written to him once. Hopefully we will get some feedback. Fingers crossed 🙂
I think either way, though, we can definitely start gathering ideas. I am grateful for any help. I'll have to read up on the differences between Jekyll and mkdocs myself. In general, it would probably be good to have the wiki code in the same repo as the UI code, but in separate directories in the root directory. 
Maybe then it would be easier to work with symlinks or have the individual pieces of information pulled together via a Github action. 
I am also still undecided whether the separate treatment of custom cards makes sense in the future or whether we should not simply distinguish between internal templates and final card templates intended for the user.
So much for that between door and door from me ;-)

Thanks again to all of you. I think it's really great how this has become a truly community driven project

**Reply by bavo** [reply]
If your going to say thanks to us for carrying the project, you are underestimating the work you're putting into this project 😛 
@CM000n Thanks for keeping an eye on everything that moves in this repository ❤️

**Reply by stokkie90** [reply]
Some cool feature of mkdocs: https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#embedding-external-files

**Reply by CM000n** [reply]
To be honest, that doesn't mean anything to me for now 😄 But sure, if mkdocs offers more features and is just as or easier to maintain than jekyll, we can gladly change the static site generator as well.

**Reply by CM000n** [reply]
I had contact with Github support again today. 
Github will now, on behalf of all of us, send @Paddy0174 the following message.

```
Dear Patrick,

we really miss you very much and hope that everything is all right with you.
During your absence, time has not stood still and we have implemented some cool new features & bug fixes and got some more things rolling for the future of Minimalist UI.
Apart from the fact that working on the project is only half as much fun without you, we also have no access to the organization and the associated repo of our wiki page without you.
We are in a dichotomy with everything we are doing right now. The Lovelace Minimalist UI is largely your baby and we don't want to ignore you in our decisions. 
On the other hand, innovations are constantly taking place in the Homeassistant environment, making it necessary to constantly adapt the UI.
If we don't hear from you in the near future despite repeated attempts to contact you, we may have to create a new organization and switch to a new repo. :-( 
This decision is difficult for us and we want to avoid it at all costs, but we hope you understand if it should come to that.

Currently the repo is mainly maintained by CM000n, schumijo, TBens and bavo.
Please contact us in the Home Assistant forum if you can.
We miss you.

Your maintainers and lovers of the UI-Lovelace-Minimalist project
```

**Reply by stokkie90** [reply]
Let's Hope he will see the message. 

I was thinking as we have gathered some people that have some great ideas to move forward, might it be an idea to create like a discord server so we can chat about the ideas we have? In case you like the idea. 

Is not much yet but: https://discord.gg/aygsyzeh

**Reply by CM000n** [reply]
Cool idea, I'm definitely in! I think I'll drop in there from time to time too. Just not as regularly as usual over the holidays.
